### ISSUE TITLE
ElasticSearch contener goes down after start

### ISSUE THREAD

**Issue by bamburillo** [markshust/docker-magento#263]
**Description**
On fresh installation then all conteiners is up  only ElasticSearch is dropped. 

Here the log 
`{"type": "server", "timestamp": "2020-05-11T19:51:15,663Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "124de8f05b65", "message": "loaded plugin [analysis-icu]" }
{"type": "server", "timestamp": "2020-05-11T19:51:15,664Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "124de8f05b65", "message": "loaded plugin [analysis-phonetic]" }
{"type": "server", "timestamp": "2020-05-11T19:51:20,339Z", "level": "WARN", "component": "o.e.c.s.SettingsModule", "cluster.name": "docker-cluster", "node.name": "124de8f05b65", "message": "\n*************************************************************************************\nFound index level settings on node level configuration.\n\nSince elasticsearch 5.x index level settings can NOT be set on the nodes \nconfiguration like the elasticsearch.yaml, in system properties or command line \narguments.In order to upgrade all indices the settings must be updated via the \n/${index}/_settings API. Unless all settings are dynamic all indices must be closed \nin order to apply the upgradeIndices created in the future should use index templates \nto set default values. \n\nPlease ensure all required values are updated on all indices by executing: \n\ncurl -XPUT 'http://localhost:9200/_all/_settings?preserve_existing=true' -d '{\n  \"index.max_shingle_size\" : \"2\",\n  \"index.min_shingle_size\" : \"2\"\n}'\n*************************************************************************************\n" }
{"type": "server", "timestamp": "2020-05-11T19:51:20,349Z", "level": "ERROR", "component": "o.e.b.ElasticsearchUncaughtExceptionHandler", "cluster.name": "docker-cluster", "node.name": "124de8f05b65", "message": "uncaught exception in thread [main]",
"stacktrace": ["org.elasticsearch.bootstrap.StartupException: java.lang.IllegalArgumentException: node settings must not contain any index level settings",
"at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:174) ~[elasticsearch-7.6.2.jar:7.6.2]",
"at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:161) ~[elasticsearch-7.6.2.jar:7.6.2]",
"at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) ~[elasticsearch-7.6.2.jar:7.6.2]",
"at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125) ~[elasticsearch-cli-7.6.2.jar:7.6.2]",
"at org.elasticsearch.cli.Command.main(Command.java:90) ~[elasticsearch-cli-7.6.2.jar:7.6.2]",
"at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:126) ~[elasticsearch-7.6.2.jar:7.6.2]",
"at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92) ~[elasticsearch-7.6.2.jar:7.6.2]",
"Caused by: java.lang.IllegalArgumentException: node settings must not contain any index level settings",
"at org.elasticsearch.common.settings.SettingsModule.<init>(SettingsModule.java:143) ~[elasticsearch-7.6.2.jar:7.6.2]",
"at org.elasticsearch.node.Node.<init>(Node.java:356) ~[elasticsearch-7.6.2.jar:7.6.2]",
"at org.elasticsearch.node.Node.<init>(Node.java:257) ~[elasticsearch-7.6.2.jar:7.6.2]",
"at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.6.2.jar:7.6.2]",
"at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.6.2.jar:7.6.2]",
"at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.6.2.jar:7.6.2]",
"at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170) ~[elasticsearch-7.6.2.jar:7.6.2]",
"... 6 more"] }
uncaught exception in thread [main]
java.lang.IllegalArgumentException: node settings must not contain any index level settings
	at org.elasticsearch.common.settings.SettingsModule.<init>(SettingsModule.java:143)
	at org.elasticsearch.node.Node.<init>(Node.java:356)
	at org.elasticsearch.node.Node.<init>(Node.java:257)
	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221)
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349)
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170)
	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:161)
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125)
	at org.elasticsearch.cli.Command.main(Command.java:90)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:126)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92)
For complete error details, refer to the log at /usr/share/elasticsearch/logs/docker-cluster.log
`

**Steps To Reproduce**
Just install and check containers va `bin/status`

```
          Name                        Command               State                                              Ports
---------------------------------------------------------------------------------------------------------------------------------------------------------------
magento2_app_1             nginx -g daemon off;             Up       80/tcp, 0.0.0.0:80->8000/tcp, 0.0.0.0:443->8443/tcp
magento2_db_1              /docker-entrypoint.sh mysqld     Up       0.0.0.0:3306->3306/tcp
magento2_elasticsearch_1   /usr/local/bin/docker-entr ...   Exit 1
magento2_mailhog_1         MailHog                          Up       0.0.0.0:32768->1025/tcp, 0.0.0.0:8025->8025/tcp
magento2_phpfpm_1          docker-php-entrypoint php-fpm    Up       9000/tcp, 9001/tcp
magento2_rabbitmq_1        docker-entrypoint.sh rabbi ...   Up       15671/tcp, 0.0.0.0:15672->15672/tcp, 25672/tcp, 4369/tcp, 5671/tcp, 0.0.0.0:5672->5672/tcp
magento2_redis_1           docker-entrypoint.sh redis ...   Up       6379/tcp
```

**Comment by UglyWillDuckling** [comment]
Can confirm the issue. It only happens when using 7.6.2-1 image version.
Falling back to 7.6.2-0 removes this error, but instead of that there is an error when running the reindexing. Not sure if this new error is caused by Magento itself or by something outside.

**Comment by markshust** [comment]
thanks, able to replicate. i'm looking for a solution.
```
elasticsearch_1  | {"type": "server", "timestamp": "2020-05-12T19:31:19,251Z", "level": "WARN", "component": "o.e.c.s.SettingsModule", "cluster.name": "docker-cluster", "node.name": "49c42b4b1313", "message": "\n*************************************************************************************\nFound index level settings on node level configuration.\n\nSince elasticsearch 5.x index level settings can NOT be set on the nodes \nconfiguration like the elasticsearch.yaml, in system properties or command line \narguments.In order to upgrade all indices the settings must be updated via the \n/${index}/_settings API. Unless all settings are dynamic all indices must be closed \nin order to apply the upgradeIndices created in the future should use index templates \nto set default values. \n\nPlease ensure all required values are updated on all indices by executing: \n\ncurl -XPUT 'http://localhost:9200/_all/_settings?preserve_existing=true' -d '{\n  \"index.max_shingle_size\" : \"2\",\n  \"index.min_shingle_size\" : \"2\"\n}'\n*************************************************************************************\n" }
```

**Comment by markshust** [comment]
All fixed in `markoshust/magento-elasticsearch:7.6.2-2` (building now - give it a few). 

I backed out the updates to `elasticsearch.yml`, I tried to implement a fix for somebody using that but apparently that errored out. 

If you wish to update a setting in elasticsearch, you must do so with curl commands. For example, to update the `magento` index settings, do something like so:

```
docker-compose exec elasticsearch curl -XPUT 'http://localhost:9200/magento' -H 'Content-Type: application/json' -d '{"settings": {"index": {"max_shingle_diff": 10 }}}'
```
